### PR TITLE
Added basic eventing.

### PR DESCRIPTION
**Pull request by mrtrumbe**
Added basic storage events which are emitted via node's events module when setItem and removeItem are called.

I tried to not mess up the current performance for those not using events by checking the listenerCount and not doing extra work if no events are needed.

### CODE REVIEW

File: LocalStorage.coffee [hunk 1]
```diff
@@ -92,6 +101,9 @@ class LocalStorage
       @metaKeyMap[key] = metaKey
       @length += 1
       @bytesInUse += valueStringLength
+    if hasListeners
+      evnt = new StorageEvent key, oldValue, value
```
Comment (lmaccherone) [inline LocalStorage.coffee]: Should this have `filename` added to the last @url parameter?

Also, I know it's not canonical CoffeeScript, but could you please add parens to this (and other function calls). That's my personal style guide for CoffeeScript.

File: LocalStorage.coffee [hunk 2]
```diff
@@ -115,12 +127,19 @@ class LocalStorage
...
+      if hasListeners
+        oldValue = this.getItem(key)
       delete @metaKeyMap[key]
       @length -= 1
       @bytesInUse -= metaKey.size
       filename = path.join(@location, metaKey.key)
       @keys.splice(metaKey.index,1)
       _rm(filename)
+      if hasListeners
+        evnt = new StorageEvent key, oldValue, null
```
Comment (lmaccherone) [inline LocalStorage.coffee]: send `filename` in for the @url parameter?

File: test/LocalStorageTest.coffee [hunk 3]
```diff
@@ -79,3 +79,41 @@ exports.LocalStorageTest =
...
+      url = evnt.url
+
+    localStorage.on('storage', handleEvent)
+    
+    localStorage.setItem('a', 'something')
+    test.equal(localStorage.getItem('a'), 'something')
+    test.equal(key, 'a')
+    test.equal(oldVal, null)
+    test.equal(newVal, 'something')
+    test.equal(url, '')
```
Comment (lmaccherone) [inline test/LocalStorageTest.coffee]: You'll have to filter out the beginning parts of the url when you put the filename in here.

File: LocalStorage.coffee [hunk 4]
```diff
@@ -115,12 +127,19 @@ class LocalStorage
...
+        this.emit 'storage', evnt
     
   key: (n) ->
```
Comment (lmaccherone) [inline LocalStorage.coffee]: What about an event on `clear()`?

**Comment by lmaccherone** [comment]
I've looking into events several times before for node-localstorage. My fear is that it is primarily designed to throw events in other windows/tabs/browsers connected to the same localStorage. So, to implement that correctly, we'd have to assume other node.js instances were using the same directory and we'd have to monitor the file system to sense when an item (file) changed.

Look here:
http://stackoverflow.com/questions/4679023/bug-with-chromes-localstorage-implementation/4679754#4679754

Note, that restriction that events don't fire in the current window is not implemented in all browsers. It seems to be fairly common to fire all the time although it's not universal.

For that reason and because it sounds like it would be useful for you, I'm willing to merge it in if it matches the spec (http://dev.w3.org/html5/webstorage/#localStorageEvent). It looks like it matches the spec pretty good. I think we should use `"pid:" + process.pid` for the url parameter in case we later add filesystem monitoring.

One more suggestion, can you please add events for the `clear()` function call?

Also, please add parens to your function calls. I know it's not canonical CoffeeScript but it's my personal style guide.

**Comment by mrtrumbe** [comment]
That all sounds good to me and I'll make your suggested changes later today. My current use case is in a single node instance, so I hadn't considered how this might work with multiple node instances. Your suggestion and assumptions for that eventual implementation seem reasonable to me. 

Also, thanks for considering this!

**Comment by lmaccherone** [comment]
I also noticed that you are missing the `storageArea` parameter in the event. I think it's just a distinction between local versus session so I'm not sure how valuable it is, but it's another thing we should call out when we update the docs, unless we implement it.

BTW, thanks a ton for your contribution.

**Comment by lmaccherone** [comment]
Actually, it should be easy enough to just hard-code `storageArea` to `'localStorage'`

**Pull request merged**